Lib/importlib/metadata.py

```diff
@@ -102,16 +104,24 @@ def extras(self):
         match = self.pattern.match(self.value)
         return list(re.finditer(r'\w+', match.group('extras') or ''))
 
+    @property
+    def distribution(self):
+        if self._distribution is None:
+            self._distribution = Distribution.from_name(self.dist)
+        return self._distribution
+
```

Comment:
Assuming we keep the late resolution of the distribution, I'd use this syntax:

```suggestion
    @cached_property
    def distribution(self):
        return Distribution.from_name(self.dist)

```

Reply:
Yes, make sense.